metis-normalization/src/test/java/eu/europeana/normalization/normalizers/CleanIRIViolationsNormalizerTest.java

```diff
@@ -0,0 +1,89 @@
+package eu.europeana.normalization.normalizers;
+
+import org.apache.jena.iri.Violation;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+import java.util.stream.StreamSupport;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class CleanIRIViolationsNormalizerTest {
+
+  private final CleanIRIViolationsNormalizer normalizer = new CleanIRIViolationsNormalizer();
+  private static final String WHITESPACE_VIOLATION = "http://example.com/query?q=random word";
+  private static final String UNWISE_CHARACTER_VIOLATION = "http://example.com/query?q=random<word";
+  private static final String ILLEGAL_CHARACTER_VIOLATION = "ht$tp://example.com/query?q=random_word";
+  private static final String NOT_XML_SCHEMA_WHITESPACE_VIOLATION = "http://example.com/query?q=random\u0009word";
+  private static final String DOUBLE_WHITESPACE_VIOLATION = "http://example.com/query?q=random  word";
+  private static final String CONTROL_CHARACTER_VIOLATION = "http://example.com/query?q=random\u0085word";
+  private static final String COMPATIBILITY_CHARACTER_VIOLATION = "http://example.com/query?q=randomໜword";
+
+
+  @Test
+  public void whitespaceViolationCleanupTest(){
+    List<NormalizedValueWithConfidence> result = normalizer.normalizeValue(WHITESPACE_VIOLATION);
+    Iterator<Violation> violations = normalizer.getViolations();
+    assertTrue(StreamSupport.stream(Spliterators.spliteratorUnknownSize(violations, Spliterator.SORTED), false)
```

Comment:
How do you know that the list is SORTED? Not that it is important in this case but I would probably use another characteristic..
Also all the tests are having one violation, since in the test this is what is expected you can directly access the first violation by violations.next().codeName(), in that way you win some processing time here, what do you think?

Reply:
I actually did initially violations.next().codeName(), but for some of the test cases, there are more than 1 violation besides the one we want to test, so I decided to change to what it shows now.

Reply:
I see, makes sense! I'll leave it up to you to decide here then.
Same goes for the Spliterator.SORTED, as I said not crucial. If we know it's sorted then fine otherwise perhaps a SIZED or ORDER is more relevant?

Reply:
Indeed, either SIZED or ORDER seem more relevant, will change it